src/expr/term_canonize.cpp

```diff
@@ -107,13 +106,22 @@ Node TermCanonize::getCanonicalFreeVar(TypeNode tn, unsigned i)
     std::stringstream os;
     os << typ_name[0] << i;
     Node x = nm->mkBoundVar(os.str().c_str(), tn);
-    InstVarNumAttribute ivna;
-    x.setAttribute(ivna, d_cn_free_var[tn].size());
+    d_fvIndex[x] = d_cn_free_var[tn].size();
     d_cn_free_var[tn].push_back(x);
   }
   return d_cn_free_var[tn][i];
 }
 
+size_t TermCanonize::getIndexForFreeVariable(Node v) const
+{
+  std::map<Node, size_t>::const_iterator it = d_fvIndex.find(v);
+  if (it == d_fvIndex.end())
+  {
+    return 0;
+  }
+  return it->second;
```

Comment:
don't you wanna inline this?

Reply:
I think the compiler takes care of these low level issues.